### ISSUE TITLE
I just want to ask a question, why user Where().ToArray().FirstOrDefault() not Where().FirstOrDefault();

### ISSUE THREAD

**Issue by jessejiang0214** [Azure/azure-documentdb-dotnet#249]
Hi,

Here is the code,
https://github.com/Azure/azure-documentdb-dotnet/blob/d17c0ca5be739a359d105cf4112443f65ca2cb72/samples/code-samples/ServerSideScripts/Program.cs#L554

I just want to ask a question why user ToArray before FirstOrDefault, is there any advantage of that?

Thanks

**Comment by bpatra** [comment]
You should consider posting your question on [Stackoverflow](https://stackoverflow.com)

**Comment by kirankumarkolli** [comment]
@jessejiang0214 fair point. In this specific case only one entry will be returned from server so more over the overhead is on client side. We will address it.
